src/main/java/unicash/model/transaction/predicates/TransactionCategoryContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,67 @@
+package unicash.model.transaction.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import unicash.commons.util.StringUtil;
+import unicash.commons.util.ToStringBuilder;
+import unicash.model.category.UniqueCategoryList;
+import unicash.model.transaction.Transaction;
+
+/**
+ * Tests that a {@code Transactions}'s {@code UniqueCategoryList} matches any of
+ * the keywords given.
+ */
+public class TransactionCategoryContainsKeywordsPredicate
+        implements Predicate<Transaction> {
+    private final List<String> keywords;
+
+    public TransactionCategoryContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Transaction transaction) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsSubstringIgnoreCase(
+                        categoryJoiner(transaction.getCategories()), keyword));
+    }
+
+    /**
+     * Returns all categories as a single string.
+     *
+     * @param categoryList
+     * @return
+     */
+    public String categoryJoiner(UniqueCategoryList categoryList) {
+        String originalToString = categoryList.toString();
+        String trimmedToString = originalToString
+                .substring(1, originalToString.length() - 1);
+
+        String[] trimmedToStringArray = trimmedToString.split(", ");
+        String joinedString = String.join("", trimmedToString);
+
+        return joinedString;
+    }
```

Comment:
Add on to this point, having this method here seems to be breaking cohesion since this method deals primarily with categories, might be a job for the UniqueCategoryList instead of a Predicate class

Reply:
Good idea, thanks for the suggestions guys, I've added that change